pkg/controller/workload/kubernetes/application/application.go

```diff
@@ -98,16 +98,25 @@ type localCluster struct {
 }
 
 func (c *localCluster) sync(ctx context.Context, app *v1alpha1.KubernetesApplication) (v1alpha1.KubernetesApplicationState, error) {
-	app.Status.DesiredResources = len(app.Spec.ResourceTemplates)
-	app.Status.SubmittedResources = 0
-
 	var errs []error
 
 	// Garbage collect any resource we control but no longer have templates for.
 	if err := c.gc.process(ctx, app); err != nil {
 		return v1alpha1.KubernetesApplicationStateFailed, errors.Wrap(err, errGarbageCollect)
 	}
```

Comment:
Should we return before running GC if we're being deleted? Kubernetes GC should take care of all KARs we made in this case.

Reply:
Yes! Since to be garbage collected a KAR must be owned by the KA, and if it is owned by the KA then it will be cleaned up when the KA is deleted and all finalizers are removed from the KAR. Thanks for catching that, will update :+1: